extensions/wikia/PageHeader/ActionButton.class.php

```diff
@@ -0,0 +1,131 @@
+		$this->buttonAction['data-tracking'] = $this->buttonAction['id'];
+		return $this->buttonAction;
+	}
+
+	public function getDropdownActions(): array {
+		// items to be added to "edit" dropdown
+		$actions = [
+			'edit',
+			've-edit',
+			'formedit',
+			'history',
+			'move',
+			'protect',
+			'unprotect',
+			'delete',
+			'undelete',
+			'replace-file',
+			'talk',
+			'edit-mobile-main-page'
+		];
+
+		// Enable to modify actions list on dropdown
+		wfRunHooks( 'PageHeaderDropdownActions', [ &$actions ] );
+
+		$ret = [];
+		foreach ( $actions as $action ) {
+			if ( isset( $this->contentActions[$action] ) ) {
+				$ret[$action] = $this->contentActions[$action];
+				if ( isset( $ret[$action]['id'])) {
+					$ret[$action]['data-tracking'] = $ret[$action]['id'] . '-dropdown';
+				}
+			}
+		}
+
+		return $ret;
+	}
+
+	private function prepareActionButton() {
+		global $wgTitle, $wgUser, $wgRequest, $wgEnableCuratedContentExt;
+
+		wfRunHooks( 'BeforePrepareActionButtons', [ $this, &$this->contentActions ] );
+
+		$isDiff = !is_null( $wgRequest->getVal( 'diff' ) );
+
+		// "Add topic" action
+		if ( isset( $this->contentActions['addsection'] ) ) {
+			// remove on diff pages (RT #72666)
+			if ( $isDiff ) {
+				unset( $this->contentActions['addsection'] );
+			}
+		}
+
+		// handle protected pages (they should have viewsource link and lock icon) - BugId:9494
+		if ( isset( $this->contentActions['viewsource'] ) &&
+			!$wgTitle->isProtected() &&
+			!$wgTitle->isNamespaceProtected( $wgUser ) &&
+			!$wgUser->isLoggedIn() /* VOLDEV-74: logged in users should see the viewsource button, not edit */
+		) {
+			// force login to edit page that is not protected
+			$this->contentActions['edit'] = $this->contentActions['viewsource'];
+			$this->contentActions['edit']['text'] = wfMessage( 'page-header-action-button-edit' )->escaped();
+			unset( $this->contentActions['viewsource'] );
+		}
+
+		// If cascade protected, show viewsource button - BugId:VE-89
+		if ( isset( $this->contentActions['edit'] ) && $wgTitle->isCascadeProtected() ) {
+			$this->contentActions['viewsource'] = $this->contentActions['edit'];
+			$this->contentActions['viewsource']['text'] = wfMessage( 'page-header-action-button-viewsource' )->escaped();
+			unset( $this->contentActions['edit'] );
+		}
+
+		// "Add topic"
+		if ( isset( $this->contentActions['addsection'] ) ) {
+			$action = $this->contentActions['addsection'];
+			$action['text'] = wfMessage( 'page-header-action-button-add-topic' )->escaped();
+			$this->buttonAction = $action;
+		}  // ve-edit
+		else if ( isset( $this->contentActions['ve-edit'] ) && $this->contentActions['ve-edit']['main'] ) {
+			$this->buttonAction = $this->contentActions['ve-edit'];
+			unset( $this->contentActions['ve-edit'] );
+		} // edit
+		else if ( isset( $this->contentActions['edit'] ) ) {
+			$this->buttonAction = $this->contentActions['edit'];
+			unset( $this->contentActions['edit'] );
+		} // view source
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)